internal/loggers/dbc_passive_logger.go

```diff
@@ -0,0 +1,148 @@
+package loggers
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/edge-network/internal/models"
+
+	"github.com/pkg/errors"
+
+	"github.com/DIMO-Network/edge-network/internal/loggers/canbus"
+	"github.com/rs/zerolog"
+	"golang.org/x/sys/unix"
+)
+
+//go:generate mockgen -source dbc_passive_logger.go -destination mocks/dbc_passive_logger_mock.go
+type DBCPassiveLogger interface {
+	StartScanning(ch chan<- models.SignalData) error
+	HasDBCFile() bool
+}
+
+type dbcPassiveLogger struct {
+	logger  zerolog.Logger
+	dbcFile *string
+}
+
+func NewDBCPassiveLogger(logger zerolog.Logger, dbcFile *string) DBCPassiveLogger {
+	return &dbcPassiveLogger{logger: logger, dbcFile: dbcFile}
+}
+
+func (dpl *dbcPassiveLogger) StartScanning(ch chan<- models.SignalData) error {
+	if dpl.dbcFile == nil {
+		return nil
+	}
+	filters, err := dpl.parseDBCHeaders(*dpl.dbcFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to pase dbc file: %s", *dpl.dbcFile)
+	}
+
+	recv, err := canbus.New()
+	if err != nil {
+		return err
+	}
+	defer recv.Close()
+
+	// set hardware filters
+	uf := make([]unix.CanFilter, len(filters))
+	for i, filter := range filters {
+		uf[i].Id = filter.header // wants decimal representation of header - not hex
+		uf[i].Mask = unix.CAN_SFF_MASK
+	}
+	err = recv.SetFilters(uf)
+	if err != nil {
+		return fmt.Errorf("cannot set canbus filters: %w", err)
+	}
+	err = recv.Bind("can0")
+	if err != nil {
+
+		return errors.Wrap(err, "could not bind recv socket")
+	}
+
+	// loop
+	for {
+		frame, err := recv.Recv()
```

Comment:
Would this loop run indefinitely as log as edge-network runs? Would it be querying the BBC's one by one indefinitely?

Reply:
yes, it would run indefinitely. Had been wondering if would be good to add a select case with a "stop" channel. 
I'm adding a 1 second sleep btw each loop to not be hammering the recv function as much. I would not expect these signals to pop up that often.

Reply:
I see, I think stop channel might be useful at least for tests perspective, similarly as we use stop channel in worker runner in tests

Reply:
Oh, because it is runs in separate goroutine, we do not need it, I see all tests passed anyway